azext_iot/iothub/providers/certificate.py

```diff
@@ -0,0 +1,102 @@
+
+
+logger = get_logger(__name__)
+
+
+class CertificateProvider(IoTHubProvider):
+    def __init__(
+        self,
+        cmd,
+        hub_name: Optional[str] = None,
+        rg: Optional[str] = None,
+        login: Optional[str] = None,
+        auth_type_dataplane: Optional[str] = None
+    ):
+        super(CertificateProvider, self).__init__(
+            cmd=cmd, hub_name=hub_name, rg=rg, login=login, auth_type_dataplane=auth_type_dataplane
+        )
+        self.cli = EmbeddedCLI()
+        # Need to ensure rg
+        if "resourcegroup" not in self.target:
+            self.target["resourcegroup"] = self.discovery.get_target(
+                resource_name=self.target["entity"].split(".")[0]
+            )["resourcegroup"]
+
+    def iot_hub_certificate_root_authority_show(self) -> Optional[Dict[str, str]]:
+        return self._get_target_root_certificate()
+
+    def iot_hub_certificate_root_authority_set(
+        self,
+        ca_version: str,
+        yes: bool = False
+    ) -> Optional[Dict[str, str]]:
+        current_target = self._get_target_root_certificate()
+        if not current_target:
+            return
+        root_ca = current_target.get("enableRootCertificateV2") or False
+
+        # Check if changes are needed
+        if ca_version == CertificateAuthorityVersions.v1.value and root_ca:
+            print(CA_REVERT_WARNING)
+            if not yes and not prompt_y_n(msg=CONT_INPUT_MSG, default="n"):
+                raise ManualInterrupt(ABORT_MSG)
+        elif ca_version == CertificateAuthorityVersions.v2.value and not root_ca:
+            print(CA_TRANSITION_WARNING)
+            if not yes and not prompt_y_n(msg=CONT_INPUT_MSG, default="n"):
+                raise ManualInterrupt(ABORT_MSG)
+        else:
+            print(NO_CHANGE_MSG.format(ca_version))
+            return
+
+        result = self.cli.invoke(
+            "resource update -n {} -g {} --api-version {} --set properties.rootCertificate.enableRootCertificateV2={} "
+            "--resource-type {}".format(
+                self.target["entity"].split(".")[0],
+                self.target['resourcegroup'],
+                CA_TRANSITION_API_VERSION,
+                not root_ca,
+                HUB_PROVIDER
+            )
+        )
+        if not result.success():
+            return
+        return result.as_json().get("properties").get("rootCertificate")
+
+    def _get_target_root_certificate(self) -> Optional[Dict[str, str]]:
+        result = self.cli.invoke(
+            "resource show -n {} -g {} --api-version {} --resource-type {}".format(
+                self.target["entity"].split(".")[0],
+                self.target['resourcegroup'],
+                CA_TRANSITION_API_VERSION,
+                HUB_PROVIDER
+            )
+        )
+        if not result.success():
+            # Error will already be printed out
+            return
+        current_target = result.as_json().get("properties").get("rootCertificate")
```

Comment:
tiny nit, if 'properties' will always be there, you might as well do `result.as_json()["properties"].get("rootCertificate")`. If for reasons, there is a _chance_ a properties object/dict won't be returned then we can be slightly more defensive by doing `result.as_json().get("properties", {}).get("rootCertificate")`  (providing the default dict fallback on the first get).